crates/precompile/src/modexp.rs

```diff
@@ -108,15 +103,16 @@ where
     }
 
     // Padding is needed if the input does not contain all 3 values.
-    let base = right_pad_with_offset_vec(input, 0, base_len);
-    let exponent = right_pad_with_offset_vec(input, base_len, exp_len);
-    let modulus = right_pad_with_offset_vec(input, base_len.saturating_add(exp_len), mod_len);
+    let input = right_pad_vec(input, base_len + exp_len + mod_len);
+    let (base, input) = input.split_at(base_len);
+    let (exponent, modulus) = input.split_at(exp_len);
+    debug_assert_eq!(modulus.len(), mod_len);
```

Comment:
Made this allocate only once in case input is too short.

`modulus.len()` must be `mod_len` due to the padding above, this is the same as slicing the input 3 times.